contracts/MetadataViews.cdc

```diff
@@ -377,162 +311,325 @@ pub contract MetadataViews {
     ///
-    pub struct Media {
+    access(all) struct Trait {
+        // The name of the trait. Like Background, Eyes, Hair, etc.
+        access(all) let name: String
 
-        /// File for the media
+        // The underlying value of the trait, the rest of the fields of a trait provide context to the value.
+        access(all) let value: AnyStruct
+
+        // displayType is used to show some context about what this name and value represent
+        // for instance, you could set value to a unix timestamp, and specify displayType as "Date" to tell
+        // platforms to consume this trait as a date and not a number
+        access(all) let displayType: String?
+
+        // Rarity can also be used directly on an attribute.
+        //
+        // This is optional because not all attributes need to contribute to the NFT's rarity.
+        access(all) let rarity: Rarity?
+
+        view init(name: String, value: AnyStruct, displayType: String?, rarity: Rarity?) {
+            self.name = name
+            self.value = value
+            self.displayType = displayType
+            self.rarity = rarity
+        }
+    }
+
+    /// Wrapper view to return all the traits on an NFT.
+    /// This is used to return traits as individual key/value pairs along with
+    /// some contextualized data about each trait.
+    access(all) struct Traits {
+        access(all) let traits: [Trait]
+
+        view init(_ traits: [Trait]) {
+            self.traits = traits
+        }
+            
+        /// Adds a single Trait to the Traits view
+        /// 
+        /// @param Trait: The trait struct to be added
         ///
-        pub let file: AnyStruct{File}
+        access(all) fun addTrait(_ t: Trait) {
+            self.traits.append(t)
+        }
+    }
 
-        /// media-type comes on the form of type/subtype as described here
-        /// https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types
+    /// Helper to get Traits view in a typesafe way
+    ///
+    /// @param viewResolver: A reference to the resolver resource
+    /// @return A optional Traits struct
+    ///
+    access(all) fun getTraits(_ viewResolver: &{ViewResolver.Resolver}) : Traits? {
+        if let view = viewResolver.resolveView(Type<Traits>()) {
+            if let v = view as? Traits {
+                return v
+            }
+        }
+        return nil
+    }
+
+    /// Helper function to easily convert a dictionary to traits. For NFT 
+    /// collections that do not need either of the optional values of a Trait, 
+    /// this method should suffice to give them an array of valid traits.
+    ///
+    /// @param dict: The dictionary to be converted to Traits
+    /// @param excludedNames: An optional String array specifying the `dict`
+    ///         keys that are not wanted to become `Traits`
+    /// @return The generated Traits view
+    ///
+    access(all) fun dictToTraits(dict: {String: AnyStruct}, excludedNames: [String]?): Traits {
```

Comment:
This function isn't going to work anymore from where it is used most (`getViews`), we will need to rework it a bit looks like

Reply:
This isn't used from `getViews`, is it? Isn't it used from `resolveView`?